packages/salesforcedx-vscode-core/src/util/orgUtil.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Aliases } from '@salesforce/core';
+import { OUTPUT_CHANNEL } from '../channels';
+import { nls } from '../messages';
+import { notificationService, ProgressNotification } from '../notifications';
+import { FileInfo, OrgList } from '../orgPicker';
+
+export async function setUpOrgExpirationWatcher() {
+  // Run once to start off with.
+  await checkForExpiredOrgs();
+
+  /*
+  Comment this out for now.  For now, we are only going to check once on activation,
+  however it would be helpful if we also checked once a day.  If we decide to also
+  check once a day, uncomment the following code.
+
+  // And run again once every 24 hours.
+  setInterval(async () => {
+    await checkForExpiredOrgs();
+  }, 1000 * 60 * 60 * 24);
+  */
+}
+
+export async function checkForExpiredOrgs() {
+  try {
+    const daysBeforeExpire = 5;
```

Comment:
Is this something we could put behind a custom setting, and maybe allow for no reminder? I usually made my scratch orgs last 7 days (some folks I know make theirs only for 1-2), so it would be a bit more frequent/intrusive. Also cool if that's a better fit for a subsequent item 👍

Reply:
I think it would be good for a followup item.  Let's see how the users like this feature, and how they use it.  If we add the "also check daily" feature in, it would be good then to revisit this and maybe add a custom setting.  I think for right now, I'd like to not make all the options user configurable (too many options = too many choices for the user)